src/models/Experiments.php

```diff
@@ -301,7 +301,14 @@ public function getSteps(): array
         // clean up the results so we get a nice array with experiment id/title and steps with their id/body
         // use reference to edit in place
         foreach ($res as &$exp) {
-            $exp['steps'] = array_combine(explode('|', $exp['steps_id']), explode('|', $exp['steps_body']));
+            $stepIDs = explode('|', $exp['steps_id']);
+            $stepsBodies = explode('|', $exp['steps_body']);
+
+            $expSteps = array();
+            foreach ($stepIDs as $key => $stepID) {
```

Comment:
Not a big fan of a foreach inside another foreach, but I don't really see a way around it for now.